### PR TITLE
FLARE-1427 - Node process metric updates

### PR DESCRIPTION
**Pull request by andruwm**
# Overview
This PR makes a significant refactor of the node process metrics for better insights an accuracy

1. The in-house version of event loop lag was replaced with the official one exported by node which offers more metrics and a more complete picture
2. ELU is now automatically enabled along with the other node metrics
3. log messages have been updated in order to make them play nicer with DD's unit tagging and dashboards

### CODE REVIEW

File: lib/metrics.ts [hunk 1]
```diff
@@ -1,91 +1,89 @@
...
+  frequency_ms: number = 30_000
 ) => {
   const log = logger(source);
 
-  setInterval(() => log_memory_usage(log), frequency_ms);
-  start_pause_detector(log, 100, pause_threshold_ms);
-  setInterval(() => log_pauses(log), frequency_ms);
+  log_memory_usage(log, frequency_ms);
+  log_event_loop_utilization(log, frequency_ms);
+  log_event_loop_lag(log, frequency_ms);
```
Comment (taylor-sutton) [inline lib/metrics.ts:36]: Nit: I would suggest naming these functions to indicate that they don't just log once, they actually start an interval that logs repeatedly (`start_logging_*` e.g.)

File: lib/metrics.ts [hunk 2]
```diff
@@ -1,91 +1,89 @@
...
-  const logger = new kv.logger(source);
+function log_event_loop_utilization(log: metricLogger, frequency_ms: number) {
+  let last: EventLoopUtilization;
   setInterval(() => {
-    const { idle, active, utilization } = performance.eventLoopUtilization();
-    logger.infoD("event-loop-utilization", {
-      idle,
-      active,
-      utilization,
+    last = performance.eventLoopUtilization(last);
```
Comment (taylor-sutton) [inline lib/metrics.ts:54]: When `last` is `null`, does this do something reasonable?

**Review by taylor-sutton — APPROVED** [review]
Approved with questions. We should also take a follow-up task to add versioning and NPM-publishing of this.

File: lib/metrics.ts (hunk 1 above)
Comment (andruwm) [reply lib/metrics.ts]: Fixed in [7a88408](https://github.com/Clever/node-process-metrics/pull/32/commits/7a88408a79f6e2beca67a07f299f69fc90c70bc7)

**Pull request merged**